File: Stars Pattern/Diamond/Pattern.py
#!/usr/bin/env python3

"""
Diamond Pattern
"""

from sys import path

path.append("../.")


from ConsoleNumInput import inInt


#### Begin ####

n = inInt().getInput(5)

# Method 1
print("Method 1")

for a in range(n - 1): 
    
    for b in range(a, n):
        print("   ", end="")
    
    for c in range(a):
        print(" * ", end="")
    
    for d in range(a + 1):
        print(" * ", end="")
    
    print("")


for e in range(n):
    
    for f in range(e + 1):
        print("   ", end="")
    
    for g in range(e , n - 1):
        print(" * ", end="")
    
    for h in range(e, n):
        print(" * ", end="")
    
    print("")


print("\n*~*~*~*~*~*~*~*~*~*~*~*\n")


# Method 2
print("Method 2")

for a in range(n - 1):
    
    print("   " * (n - a), end="")
    
    print(" * " * a, end="")
    
    print(" * " * (a + 1))


for b in range(n):
    
    print("   " * (b + 1), end="")
    
    print(" * " * (n - b - 1), end="")
    
    print(" * " * (n - b))
    
print("")

"""
Author: Jovan De Guia
Github Username: jxmked
"""

File: Stars Pattern/Diamond/Pattern.c
#include <stdio.h>
#include <stdlib.h>

// External File
#include "../ConsoleNumInput.c"

/**
 * Diamond Pattern
 * */

static int loop(int n);


int main(void){
    // Default
    int n = 5;
    
    do {
        // From External File
        clrscr();
        
        loop(n);
        
        n = inInt();
        
        // This will never trigger. 
        // System required. Hahaha
        if(! n){
            return 0;
        }
    } while(1);
}

static int loop(int n){
    int i, j;
    
    for(i = 0; i < n; i++){
        for(j = (n - (i + 1)); j > 0; j--){
            printf("   ");
        }
        
        for(j = 0; j <= (i * 2); j++){
            printf(" * ");
        }
        
        printf("\n");
    }
    
    for(i = n - 2; i >= 0; i--){
        for(j = (n - (i + 1)); j > 0; j--){
            printf("   ");
        }
        
        for(j = 0; j <= (i * 2); j++){
            printf(" * ");
        }
        
        printf("\n");
    }
    return 0;
}

/**
 * Written by Jovan De Guia
 * Github Username: jxmked
 * */
